tests/posix/semaphore/prj.conf

```
@@ -0,0 +1,6 @@
+CONFIG_POSIX_API=y
+CONFIG_ZTEST=y
+
+CONFIG_POSIX_AEP_CHOICE_PSE52=y
```

Probably only the base system interfaces are needed here.

```suggestion
CONFIG_POSIX_AEP_CHOICE_BASE=y
```

See
https://docs.zephyrproject.org/latest/services/portability/posix/overview/index.html#configuration